profiles/api/oauth2.py

```diff
@@ -136,4 +138,14 @@ def token() -> Response:
     filtered_json_data = {your_key: json_data[your_key] for your_key in
                           ['access_token', 'expires_in', 'name', 'orcid', 'token_type']}
 
+    try:
+        profile = Profile.query.filter_by(orcid=filtered_json_data['orcid']).one()
+        profile.name = filtered_json_data['name']
+    except NoResultFound:
+        profile = Profile(filtered_json_data['name'], filtered_json_data['orcid'])
+
+        db.session.add(profile)
```

Comment:
asymmetry: both `Profile.query` and `db.*` ultimately access persistence, but only one is named `db` :thinking:

Reply:
Repository pattern doesn't cut it without UnitOfWork too, we'll see later when there is more duplication

Reply:
However, doesn't cover collisions? `id` is randomly generated, so it may already be in the database

Reply:
True, should add a retry. (We'll need to override the ID generator in a test to do that.)